packages/one-app-hmr/bin/one-app-hmr.js

```diff
@@ -0,0 +1,29 @@
+#! /usr/bin/env node
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+const {
+  getHMRConfig,
+  hmrServer,
+  preloadOneAppStatics,
+  preBuildExternals,
+} = require('../lib');
+
+// TODO: add support for cli options, merge with default config
+(async function oneAppHMR() {
+  const config = await getHMRConfig();
+  await preloadOneAppStatics(config);
+  await preBuildExternals(config);
```

Comment:
We build a DLL for all the providedExternals and requiredExternals before we start, since we need a two step build for the DLL manifest to be available for the Holocron modules build. DLLs will help reduce build times. in general.